### PR TITLE
Bump shadow to 7.1

### PR DESCRIPTION
**Pull request by Goooler**
## Description

[Shadow 7.0](https://github.com/johnrengelman/shadow/releases/tag/7.0.0) for Gradle 7.0+.

## Checklist

- [ ] tests are added
- [x] `CHANGELOG.md` is updated

### CODE REVIEW

**Comment by Goooler** [comment]
```
A problem occurred configuring project ':ktlint'.
> Dependency verification failed for configuration ':ktlint:classpath'
  17 artifacts failed verification:
    - apache-10.pom (org.apache:apache:10) from repository MavenRepo
    - apache-23.pom (org.apache:apache:23) from repository MavenRepo
    - commons-io-2.8.0.jar (commons-io:commons-io:2.8.0) from repository MavenRepo
    - commons-io-2.8.0.pom (commons-io:commons-io:2.8.0) from repository MavenRepo
    - commons-parent-52.pom (org.apache.commons:commons-parent:52) from repository MavenRepo
    - jdependency-2.6.0.jar (org.vafer:jdependency:2.6.0) from repository MavenRepo
    - jdependency-2.6.0.pom (org.vafer:jdependency:2.6.0) from repository MavenRepo
    - logging-parent-3.pom (org.apache.logging:logging-parent:3) from repository MavenRepo
    - maven-3.0.4.pom (org.apache.maven:maven:3.0.4) from repository MavenRepo
    - maven-model-3.0.4.jar (org.apache.maven:maven-model:3.0.4) from repository MavenRepo
    - maven-model-3.0.4.pom (org.apache.maven:maven-model:3.0.4) from repository MavenRepo
    - maven-parent-21.pom (org.apache.maven:maven-parent:21) from repository MavenRepo
    - plexus-5.1.pom (org.codehaus.plexus:plexus:5.1) from repository MavenRepo
    - plexus-utils-3.3.0.jar (org.codehaus.plexus:plexus-utils:3.3.0) from repository MavenRepo
    - plexus-utils-3.3.0.pom (org.codehaus.plexus:plexus-utils:3.3.0) from repository MavenRepo
    - plugin-publish-plugin-0.14.0.jar (com.gradle.publish:plugin-publish-plugin:0.14.0) from repository Gradle Central Plugin Repository
    - plugin-publish-plugin-0.14.0.pom (com.gradle.publish:plugin-publish-plugin:0.14.0) from repository Gradle Central Plugin Repository
  If the artifacts are trustworthy, you will need to update the gradle/verification-metadata.xml file by following the instructions at https://docs.gradle.org/7.2/userguide/dependency_verification.html#sec:troubleshooting-verification
```

**Comment by Goooler** [comment]
Had added hash values in `Update verification-metadata.xml` bug not work, how to solve this?

**Comment by Tapchicoma** [comment]
@Goooler open failed job and download build artifacts - there should be an archive with verification report. As far as I see - several signing keys are missing and two checksums.

How you've regenerated verification metadata?

**Comment by Goooler** [comment]
[7.1.0](https://github.com/johnrengelman/shadow/releases/tag/7.1.0) is out.

I run `./gradlew --write-verification-metadata pgp,sha256 help` again, it's ok now.

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
-   xsi:schemaLocation="https://schema.gradle.org/dependency-verification https://schema.gradle.org/dependency-verification/dependency-verification-1.0.xsd">
+<verification-metadata xmlns="https://schema.gradle.org/dependency-verification" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="https://schema.gradle.org/dependency-verification https://schema.gradle.org/dependency-verification/dependency-verification-1.1.xsd">
    <configuration>
       <verify-metadata>true</verify-metadata>
       <verify-signatures>true</verify-signatures>
       <key-servers>
-          <key-server uri="https://keys.openpgp.org"/>
-          <key-server uri="https://keyserver.ubuntu.com"/>
-          <!-- key server below is added from the https://sks-keyservers.net pool that is not working anymore -->
-          <!-- remove it when it will be possible -->
```
Comment (Tapchicoma) [inline gradle/verification-metadata.xml:13]: Please restore this comment

File: gradle/verification-metadata.xml [hunk 2]
```diff
@@ -227,17 +242,14 @@
...
       </component>
-      <component group="com.github.jengelman.gradle.plugins" name="shadow" version="6.1.0">
-         <artifact name="shadow-6.1.0.jar">
-            <sha256 value="b66cb33a1d204ffaa1ba67393bdddbe9ff517f24f4438d11c341423868759aa3" origin="Generated by Gradle because artifact wasn't signed"/>
-         </artifact>
-         <artifact name="shadow-6.1.0.pom">
-            <sha256 value="d40c29bce31762b6c8539a87d2515324f44db9d7d579a5aa7016a15ce164abb8" origin="Generated by Gradle because artifact wasn't signed"/>
+      <component group="com.github.johnrengelman.shadow" name="com.github.johnrengelman.shadow.gradle.plugin" version="7.0.0">
+         <artifact name="com.github.johnrengelman.shadow.gradle.plugin-7.0.0.pom">
+            <sha256 value="424b7d1343be0e521eec422c1e16b6be6b28042512673e4aa8499c5038e1a509" origin="Generated by Gradle"/>
```
Comment (Tapchicoma) [inline gradle/verification-metadata.xml:247]: Seems could be removed

File: gradle/verification-metadata.xml [hunk 3]
```diff
@@ -396,6 +408,22 @@
...
          </artifact>
       </component>
+      <component group="gradle.plugin.com.github.jengelman.gradle.plugins" name="shadow" version="7.0.0">
+         <artifact name="shadow-7.0.0.jar">
+            <sha256 value="8e7b12a0a2c2d170884a860679a019bab60b63bc99ebfa0db0897d249d86f345" origin="Generated by Gradle"/>
+         </artifact>
+         <artifact name="shadow-7.0.0.pom">
+            <sha256 value="6812f26f6d81eccced5411603475cb3a92594fa48077c75eb08d05165ebbf475" origin="Generated by Gradle"/>
+         </artifact>
+      </component>
```
Comment (Tapchicoma) [inline gradle/verification-metadata.xml:418]: Same here - seems could be removed

**Review by Tapchicoma — COMMENTED** [review]
Few minor issues, otherwise looks good to me

File: gradle/verification-metadata.xml (hunk 1 above)
Comment (Goooler) [reply gradle/verification-metadata.xml:13]: Reverted.

**Review by Tapchicoma — APPROVED** [review]

**Pull request merged**